src/Hedgehog/Gen.fs

```diff
@@ -162,30 +162,15 @@ module Gen =
     //
 
     /// Generates a random number in the given inclusive range.
-    let inline range (lo : ^a) (hi : ^a) : Gen<'a> =
-        create (Shrink.towards lo) (Random.range lo hi)
+    let inline integral (range : Range<'a>) : Gen<'a> =
+        create (Shrink.towards <| Range.origin range) (Random.integral range)
 
     /// Generates a random number from the whole range of the numeric type.
     let inline bounded () : Gen<'a> =
-        let lo = minValue ()
-        let hi = maxValue ()
         let zero = LanguagePrimitives.GenericZero
+        let range = Range.constantBounded ()
 
-        create (Shrink.towards zero) (Random.range lo hi)
-
-    /// Generates a random number in the given inclusive range, but smaller
-    /// numbers are generated more often than bigger ones.
-    let inline sizedRange (lo : ^a) (hi : ^a) : Gen<'a> =
-        create (Shrink.towards lo) (Random.sizedRange lo hi)
-
-    /// Generates a random number from the whole range of the numeric type, but
-    /// smaller numbers are generated more often than bigger ones.
-    let inline sizedBounded () : Gen<'a> =
-        let lo = minValue ()
-        let hi = maxValue ()
-        let zero = LanguagePrimitives.GenericZero
-
-        create (Shrink.towards zero) (Random.sizedRange lo hi)
+        create (Shrink.towards zero) (Random.integral range)
 
```

Comment:
Yeah anything to do with sized/bounded is really covered by `Range` now and shouldn't be here, perhaps with the exception of the `char` combinators because I'm not sure if they can be done with `Range`?

Of course we still need `sized`/`resize` here, just not the ones that generate data based on size.